File: 300/357 - Count Numbers with Unique Digits.c
int countNumbersWithUniqueDigits(int n) {
    int ret, s, i;
    
    if (n == 0)
        ret = 1;
    else if (n == 1)
        ret = 10;
    else {
        s = 9;
        ret = 10;
        
        for (i = 1; i < n; ++i) {
            s *= 10 - i;
            ret += s;
        }
    }
    
    return ret;
}

File: csharp/357.count-numbers-with-unique-digits.cs
/*
 * @lc app=leetcode id=357 lang=csharp
 *
 * [357] Count Numbers with Unique Digits
 */
public class Solution {

    private static readonly int[] Answers = new int[] {
        1, 10, 91, 739, 5275, 32491, 168571, 712891, 2345851, 5611771, 8877691
    };

    public int CountNumbersWithUniqueDigits(int n) {
        return Answers[Math.Min(10, n)];
    }
}

